control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -274,11 +274,15 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	// Return early if deleted
 	if !hostedControlPlane.DeletionTimestamp.IsZero() {
-		if err := r.destroyAWSDefaultSecurityGroup(ctx, hostedControlPlane); err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to destroy default workeer security group: %w", err)
-		}
 
 		if shouldCleanupCloudResources(r.Log, hostedControlPlane) {
+			if err := r.destroyAWSDefaultSecurityGroup(ctx, hostedControlPlane); err != nil {
+				if awsErrorCode(err) == "UnauthorizedOperation" {
```

Comment:
Is there a vendored const we could use?

It'd be nice to unit test this with an aws api mock

Reply:
There isn't that I could find. +1 on the unit test, will do as follow up